spec/services/slack_service/slack_user_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "rails_helper"
+
+describe SlackService::SlackUser do
+  let(:valid_params) {
+    {
+      id: "A00AA1AAA",
+      name: "cap",
+      real_name: "Steve Rogers",
+      first_name: "Steve",
+      tz: "America/New_York",
+      email: "steve@ombulabs.com",
+      client: SlackServiceSpec::SlackUserSpec::SlackClientMock
+    }
+  }
+
+  describe ".initialize" do
+    it "requires all expected keys" do
+      [:id, :name, :real_name, :first_name, :tz, :email, :client].map { |key|
+        expect { SlackService::SlackUser.new(valid_params.except!(key)) }.to raise_error(KeyError)
+      }
+    end
+  end
+
+  describe "#message" do
+    it "success: forwards the message to SlackService.send_message" do
+      user = SlackService::SlackUser.new(valid_params)
+      expect(SlackService).to receive(:send_message).with(valid_params[:id], "A message", valid_params[:client])
+
+      user.message("A message")
+    end
+  end
+end
+
+module SlackServiceSpec
+  module SlackUserSpec
+    class SlackClientMock
+      def self.chat_postMessage(params)
```

Comment:
This naming is odd, let's change it to `chat_post_message`

Reply:
This is mocking `Slack::Web::Client` (from the gem itself) - so we're not in control of the method signature